src/services/user/adminUnblockUser.js

```diff
@@ -1,12 +1,18 @@
 const {successResponse, errorResponse} = require('../../utils');
 const {adminUnblockUserProcess} = require('../../processes/scoring/admin-block-unblock-user');
+const UsersFunction = require('../../databases/functions/users');
 
 const adminUnblockUser = async (req, res) => {
   try {
     const {userId} = req.body;
     if (!userId) {
       return errorResponse(res, 'userId is required', 400);
     }
+    try {
+      await UsersFunction.userUnblockByAdmin(userId);
+    } catch (error) {
+      return errorResponse(res, 'Failed to unblock user by admin', 500);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a nested try-catch block which is not necessary and adds unnecessary complexity. You can handle all errors in the outer try-catch block. Also, it's better to provide more specific error messages for debugging purposes. Instead of 'Failed to unblock user by admin', you could include the actual error message from the database operation.

```diff
- 11:     try {
- 12:       await UsersFunction.userUnblockByAdmin(userId);
- 13:     } catch (error) {
- 14:       return errorResponse(res, 'Failed to unblock user by admin', 500);
- 15:     }
+ 11:     await UsersFunction.userUnblockByAdmin(userId);
```

In the outer catch block, you can differentiate between different types of errors:

```diff
-   } catch (error) {
+   } catch (error) {
+     if (error.message.includes('unblock')) {
+       return errorResponse(res, `Failed to unblock user by admin: ${error.message}`, 500);
+     }
```
This way, you will have more detailed error messages and a cleaner code structure.


<!-- This is an auto-generated comment by OSS CodeRabbit -->